plutus-pab/src/Cardano/Node/Types.hs

```diff
@@ -46,19 +81,113 @@ data MockServerConfig =
         }
     deriving (Show, Eq, Generic, FromJSON)
 
+-- | Configuration for 'Cardano.Node.Mock.blockReaper'
+data BlockReaperConfig =
+    BlockReaperConfig
+        { brcInterval     :: Second -- ^ interval in seconds for discarding blocks
+        , brcBlocksToKeep :: Int    -- ^ number of blocks to keep
+        }
+    deriving (Show, Eq, Generic, FromJSON)
+
+
+-- Logging ------------------------------------------------------------------------------------------------------------
+
+-- | Top-level logging data type for structural logging
+-- inside the Mock Node server.
+data MockServerLogMsg =
+    StartingSlotCoordination
+    | NoRandomTxGeneration
+    | StartingRandomTx
+    | KeepingOldBlocks
+    | RemovingOldBlocks
+    | StartingMockServer Int
+    | ProcessingChainEvent ChainEvent
+    | BlockOperation BlockEvent
+    | CreatingRandomTransaction
+    | FollowerMsg NodeFollowerLogMsg
+    deriving (Generic, Show, ToJSON, FromJSON)
+
+instance Pretty MockServerLogMsg where
+    pretty = \case
+        NoRandomTxGeneration      -> "Not creating random transactions"
+        StartingRandomTx          -> "Starting random transaction generation thread"
+        KeepingOldBlocks          -> "Not starting block reaper thread (old blocks will be retained in-memory forever"
+        RemovingOldBlocks         -> "Starting block reaper thread (old blocks will be removed)"
+        StartingMockServer p      -> "Starting Mock Node Server on port " <+> pretty p
+        StartingSlotCoordination  -> "Starting slot coordination thread"
+        ProcessingChainEvent e    -> "Processing chain event " <+> pretty e
+        BlockOperation e          -> "Block operation " <+> pretty e
+        CreatingRandomTransaction -> "Generating a random transaction"
+        FollowerMsg m             -> pretty m
+
+instance ToObject MockServerLogMsg where
+    toObject v = \case
+        NoRandomTxGeneration      ->  mkObjectStr "Not creating random transactions" ()
+        StartingRandomTx          ->  mkObjectStr "Starting random transaction generation thread" ()
+        KeepingOldBlocks          ->  mkObjectStr "Not starting block reaper thread (old blocks will be retained in-memory forever" ()
+        RemovingOldBlocks         ->  mkObjectStr "Starting block reaper thread (old blocks will be removed)" ()
+        StartingMockServer p      ->  mkObjectStr "Starting Mock Node Server on port " (Tagged @"port" p)
+        StartingSlotCoordination  ->  mkObjectStr "" ()
+        ProcessingChainEvent e    ->  mkObjectStr "Processing chain event" (Tagged @"event" e)
+        BlockOperation e          ->  mkObjectStr "Block operation" (Tagged @"event" e)
+        CreatingRandomTransaction ->  mkObjectStr "Creating random transaction" ()
+        FollowerMsg m             -> toObject v m
+
+-- | logging type for 'Cardano.Node.Follower'
+data NodeFollowerLogMsg =
```

Comment:
@j-mueller Maybe it would be nice to have some very single sentence descriptions for each of these *?* also for me, as i really don't actually know what they refer to/mean and i didn't want to write something too redundant.

Reply:
In prinicple yes, but the `NodeFollower` mechanism is going to be removed and replaced by the the socket protocol from `ouroboros-network` (by @raduom). So we can leave it as it is.

The idea was that each node follower periodically asks the node for new blocks since the last time they've asked. But with the socket protocol this polling is going to be replaced by the node pushing new blocks to the clients.

Reply:
I did that, but there will be issues with the semantics we give to the follower messages in tests. So what I think I will do, finally, is change the follower algebra to be more suited (stateful) to the way the socket protocol works and implement something sensible for testing, though I am not sure what exactly that would be.

Reply:
I will leave it as is for now, but I would appreciate some comments somewhere that at least broadly outline what a "node follower" is. Either in the Types module or in the Follower module itself or just _here_ if need be and i can add it somewhere later on my own.